multiuser_service_windows_test.go

```diff
@@ -0,0 +1,248 @@
+	}
+
+	// check for eventlog source
+	elog, err := eventlog.Open(name)
+	if err != nil {
+		t.Fatalf("Did not find expected eventlog source %q: %v", name, err)
+	}
+	elog.Close()
+
+	// this is unrealistic, usually the service is marked for deletion
+	// but not actually removed until reboot
+
+	// // hopefully service gets removed
+	// <-time.After(2 * time.Second)
+
+	// // verify service is removed
+	// _, err = m.OpenService(name)
+	// if err == nil {
+	// 	t.Fatalf("Found service %q after it should have been removed", name)
+	// }
+
+	// // verify eventlog is removed
+	// _, err = eventlog.Open(name)
+	// if err == nil {
+	// 	t.Fatalf("Found eventlog source %q after it should have been removed", name)
+	// }
+}
+
+type brokenWriter struct{}
+
+func (w brokenWriter) Write(bs []byte) (int, error) {
+	return -1, fmt.Errorf("broken writer is broken")
+}
+
+// func wrapTest(t *testing.T, name string, errorFunc func(*exec.ExitError)) {
+// if we are running the test
+// as opposed to wrapping the test
+// if os.Getenv(t.Name()) == "1" {
+// 	name := "generic-worker-" + slugid.Nice()
+// 	setupService(t, name, true)
+// 	defer cleanupService(t, name)
+// 	// use brokenWriter
+// 	logWriter = brokenWriter{}
+// 	// now we try runService in non-interactive mode
+// 	// which should attempt to use brokenWriter and fail
+// 	// with CANT_LOG_PROPERLY
+// 	runService(name, false)
+// 	return
+// }
+
+// // wrapping the test
+
+// // exitOnError will be called, so os.Exit() will be called
+// // so we need to run this test in a wrapper
+
+// cmd := exec.Command(os.Args[0], fmt.Sprintf("-test.run=%s", t.Name()))
+// cmd.Env = append(os.Environ(), fmt.Sprintf("%s=1", t.Name()))
+// stderrPipe, err := cmd.StderrPipe()
+// if err != nil {
+// 	t.Fatal(err)
+// }
+// stdoutPipe, err := cmd.StdoutPipe()
+// if err != nil {
+// 	t.Fatalf("Error getting StdoutPipe: %v", err)
+// }
+// if err := cmd.Start(); err != nil {
+// 	t.Fatal(err)
+// }
+// stderr, err := ioutil.ReadAll(stderrPipe)
+// if err != nil {
+// 	t.Fatal(err)
+// }
+// stdout, err := ioutil.ReadAll(stdoutPipe)
+// if err != nil {
+// 	t.Fatal(err)
+// }
+// err = cmd.Wait()
+// // if nonzero exit code, castable to ExitError
+// if exitError, ok := err.(*exec.ExitError); ok && err != nil {
+// 	// as expected, there was an error
+// 	exitCode := exitError.ExitCode()
+// 	if exitCode != int(CANT_LOG_PROPERLY) {
+// 		t.Logf("Expected runService() to exit with CANT_LOG_PROPERLY, got: %s", exitError.Error())
+// 		t.Logf("stderr: %s", stderr)
+// 		t.Logf("stdout: %s", stdout)
+// 		t.Fail()
+// 	}
+// } else {
+// 	t.Fatalf("Expected an error from %q: %v", strings.Join(cmd.Args, " "), err)
+// }
+// }
+
+func TestRunServiceWithBrokenWriter(t *testing.T) {
+	defer unprivilegedRecovery(t)
+
+	if !shouldRunAdminTests() {
+		t.Skipf("SKIP_ADMINISTRATOR_TESTS set, skipping %q", t.Name())
+	}
+
+	// if we are running the test
+	// as opposed to wrapping the test
```

Comment:
I think these comments don't apply now we commented out the test wrapping, right?